include/zephyr/arch/arc/arch.h

```diff
@@ -89,6 +89,14 @@ extern "C" {
 #define ARCH_STACK_PTR_ALIGN	4
 #endif /* CONFIG_64BIT */
 
+#if CONFIG_ISR_STACK_SIZE % ARCH_STACK_PTR_ALIGN
```

Comment:
FWIW, there is `BUILD_ASSERT` for this.

Reply:
Yep, the `BUILD_ASSERT` was disabled for MWDT toolchain previously - that's why we've used `#error` directive arch.h for that. But it's enabled now, so we can use it.

Changed for `BUILD_ASSERT` usage.